### PR TITLE
LPS-100523

### PR DESCRIPTION
**Pull request by brandizzi**
<h3><g-emoji class="g-emoji" alias="heavy_check_mark" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/2714.png">heavy_check_mark</g-emoji> ci:test:search - 49 out of 49 jobs passed in 1 hour 27 minutes 52 seconds 350 ms</h3>

https://github.com/brandizzi/liferay-portal/pull/939#issuecomment-569122400

Author: @holatuwol @brandizzi
Reviewer: @brandizzi

[Bug] Unresponsive portal on User Group reindexing
https://issues.liferay.com/browse/LPS-100523

### CODE REVIEW

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/b81f9bfed0...39262a5647

**Comment by brianchandotcom** [comment]
@achaparro lots of smoke tests failing for upgrade now?

@shuyangzhou Can you review this pattern? It seems like it should be baked into the Indexer framework itself. I'm allowing this one instance as a specific fix, but I don't like having this pattern applied everywhere which is what @holatuwol was thinking of.

@holatuwol please coordinate with @shuyangzhou Something seems off with this.

**Comment by achaparro** [comment]
@brianchandotcom I don't see upgrades failing in this pull but I noticed a lot of them failing this morning, this should have solved it:
https://github.com/brianchandotcom/liferay-portal/pull/82515

I will check tomorrow just in case there is something else.

**Comment by brandizzi** [comment]
@brianchandotcom @shuyangzhou Actually, I negotiated with @holatuwol to not apply this pattern to the Search framework for now. Originally, his pull request changed the Search Infrastructure itself. 

The problem is, we in the Search team did not have bandwidth to review his original pull request for now, since it would be way more impactful; also we have some work planned in the same direction for 2020. On top of that, @arboliveira was out of office, and he would be the best to review that. However, this is a TS ticket from October, so needed some solution.

We got to the compromise of applying it locally only so far—which paid off, it was way easier to review. Also, we will use this fix as the basis for the general solution.

So, indeed, it is a bit off but it was planned.

File: portal-impl/src/com/liferay/portlet/usersadmin/reindexer/UserReindexerImpl.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+				_executing = false;
+			}
+		}
+
+		@Override
+		public int hashCode() {
+			return _userId.hashCode();
+		}
+
+		public void ignore() {
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portlet/usersadmin/reindexer/UserReindexerImpl.java]: @brandizzi @holatuwol no one ever calls this ignore() method, so the _ignore is always false. All the readers to this method is dead code.

**Comment by shuyangzhou** [comment]
@holatuwol please come to talk to me when you are back to work.

**Comment by shuyangzhou** [comment]
The most off thing is, is this supposed to prevent duplicated User indexing within the same bulk reindex call, or acrossing different bulk reindexes calls?
If this is for duplication within the same bulk reindex, we can just check duplicated userIds.
If this is for duplication acrossing multiple bulk reindex, the global _queuedIndexerRequests "registry" won't be enough for this. Once one reindex call is done, the userid will be removed from it, the next duplicated reindex will just be called as normal.

So from what I can see, this is only preventing part of the duplicated reindex across different bulk reindexing, for that we already have com.liferay.portal.search.buffer infrastructure to eliminate duplication within the same transaction. If you are just trying to enlarge the scope to cover duplication elimination to multiple transactions. It should be done in a similar style. Meaning it should be generic and centralized, not per entity based.

**Pull request closed without merging**